r/R/dataset-scan.R

```diff
@@ -185,17 +185,36 @@ ScanTask <- R6Class("ScanTask",
 #' `data.frame`? Default `TRUE`
 #' @export
 map_batches <- function(X, FUN, ..., .data.frame = TRUE) {
-  if (.data.frame) {
-    lapply <- map_dfr
-  }
-  scanner <- Scanner$create(ensure_group_vars(X))
+  # TODO(ARROW-15271): possibly refactor do_exec_plan to return a RecordBatchReader
```

Comment:
```suggestion
  # TODO: ARROW-15271 possibly refactor do_exec_plan to return a RecordBatchReader
```

Super minor, but most of our TODOs have the semicolon and then the issue number (though there are others that do it with a space too)